### ISSUE TITLE
Unused variables are not detected when defined in the TRY EXCEPT branches

### ISSUE THREAD

**Issue by bhirsz** [MarketSquare/robotframework-robocop#1548]
TRY is not handled correctly:

```
    Log    ${var_used_only_before}    Keyword
    TRY
        ${var1}    Keyword
        ${var_used_outside1}    Keyword
        ${var_used_only_before}    Keyword
    EXCEPT     Error message*  AS  ${var2}
        ${var3}    Keyword
        ${var_used_outside2}    Keyword
    END
    Log    ${var_used_outside1}    Keyword
    Log    ${var_used_outside2}    Keyword
```

In the above example variables in the branches (EXCEPT, FINALLY, ELSE blocks) are not parsed.

Also the error variable (AS <>) is not handled. It should be locally scoped variable only.

**Comment by bhirsz** [comment]
It also applies to all related rules, such as variable overwritten before usage.
